File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/SBPacketPing.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;

import java.io.IOException;

public class SBPacketPing extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 7;
    
    public SBPacketPing(){
        super(PACKET_ID);
    }
    
    
    public long timeNanos;
    
    public SBPacketPing(long timeNanos){
        this();
        this.timeNanos = timeNanos;
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeLong(timeNanos);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        timeNanos = stream.readLong();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/SBPacketPing.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException

class SBPacketPing() : IPacket<PacketHandler?>(PACKET_ID) {
    var timeNanos: Long = 0

    constructor(timeNanos: Long) : this() {
        this.timeNanos = timeNanos
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeLong(timeNanos)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        timeNanos = stream!!.readLong()
    }

    companion object {
        const val PACKET_ID = 7
    }
}
